### ISSUE TITLE
File extension support for .glsl

### ISSUE THREAD

**Issue by rottendurian** [svenstaro/glsl-language-server#31]
I am unsure if this is something that you would want to commit to the main branch but using neovim with lspconfig I am naming my files using the convention of filename.frag.glsl. I was able to make a quick modification to the source code to make this work

```
EShLanguage find_language(const std::string& name)
{
    auto ext = fs::path(name).extension();
    if (ext == ".glsl")
        ext = fs::path(name).replace_extension().extension(); //replaces current extension with nothing and finds new file extension
    if (ext == ".vert" || ext == ".vs")
        return EShLangVertex;
    else if (ext == ".tesc")
        return EShLangTessControl;
    else if (ext == ".tese")
        return EShLangTessEvaluation;
    else if (ext == ".geom" || ext == ".gs")
        return EShLangGeometry;
    else if (ext == ".frag" || ext == ".fs")
        return EShLangFragment;
    else if (ext == ".comp")
        return EShLangCompute;
    throw std::invalid_argument("Unknown file extension!");
}

```

**Comment by svenstaro** [comment]
Is this a common convention? I'm not opposed to it if it were. Otherwise, I'm skeptical we should support something like this.

**Comment by nolanderc** [comment]
Perhaps we could allow the extensions to be customized using CLI arguments? Something like the following:

```
glslls --extension fragment:frag.glsl --extension vertex:vert.glsl ...
```

**Comment by svenstaro** [comment]
Guess so, but that's going to be annoying for people to configure in their editors. Better to decide on a bunch of defaults. I'm actually happy to merge a PR with someone just implementing the stuff from OP.

**Comment by rottendurian** [comment]
I am not sure if it is a common convention but it was the default that was being used in neovim with lsp-zero's lspconfig.

**Comment by svenstaro** [comment]
Would you like to make a PR to support this use case? Just make sure to add a bit more reasoning to the comment, a small paragraph perhaps, and it'd be fine I think.

**Comment by nolanderc** [comment]
Fixed in #32

**Comment by kroppt** [comment]
One problem I see with this implementation is that sometimes we `#include` other shader files when writing shaders. So, `.glsl` itself is often a valid extension for that use case.
